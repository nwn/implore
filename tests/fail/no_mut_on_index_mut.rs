use implore::impl_op;

#[derive(Copy, Clone, PartialEq, Eq, Debug)]
struct X;

#[impl_op]
fn index_mut(x: &mut X, y: u32) -> X {
    X
}

#[impl_op]
fn index_mut(x: X, y: u32) -> &mut X {
    &mut X
}

#[impl_op]
fn index_mut(x: &mut X, y: u32) -> &X {
    &X
}

#[impl_op]
fn index_mut(x: &X, y: u32) -> &mut X {
    &mut X
}

fn main() {}
