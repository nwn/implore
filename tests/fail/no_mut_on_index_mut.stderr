error: `index_mut` must return a mutable reference type
 --> $DIR/no_mut_on_index_mut.rs:7:36
  |
7 | fn index_mut(x: &mut X, y: u32) -> X {
  |                                    ^

error: the first operand of `index_mut` must be a mutable reference
  --> $DIR/no_mut_on_index_mut.rs:12:17
   |
12 | fn index_mut(x: X, y: u32) -> &mut X {
   |                 ^

error: `index_mut` must return a mutable reference type
  --> $DIR/no_mut_on_index_mut.rs:17:36
   |
17 | fn index_mut(x: &mut X, y: u32) -> &X {
   |                                    ^^

error: the first operand of `index_mut` must be a mutable reference
  --> $DIR/no_mut_on_index_mut.rs:22:17
   |
22 | fn index_mut(x: &X, y: u32) -> &mut X {
   |                 ^^
